app/models/date_validation.rb

```diff
@@ -28,11 +28,14 @@ def pre_validate_date_attribute(attribute, date)
       begin
         # Rails will cast the year part of the date to 0 if the year input parameter is a non-numeric string
         # This only seems to happen to the year part, other parts remain as strings
-        raise TypeError if (date[1]).zero?
+        raise TypeError if date[1].zero?
+
+        # Rails does not accept negative month values, but the Date constructor does
+        raise TypeError if (date[2]).negative?
```

Comment:
does the date constructor handle negative values for date[1] and date[3] then?

Reply:
Well, a negative year is legitimate (BC years), and yes it does prevent negative days: https://docs.ruby-lang.org/en/master/Date.html#method-c-new

Reply:
cool thanks. I suspect the person putting in negative values was purposefully doing to try and break it 😂 

LGTM!